pkg/pool/pool.go

```diff
@@ -715,6 +736,73 @@ func (p *pool) getPDDeviation() int {
 	return deviationPrefix * NumIPv4AddrPerPrefix
 }
 
+// calculateSecondaryIPDeviation calculates the deviation required to meet the desired state for secondary IP mode
+// Returns a number of IPv4 addresses by taking into account the MinIPTarget and WarmIPTarget
+func (p *pool) calculateSecondaryIPDeviation() int {
+	numWarmResources := numResourcesFromMap(p.warmResources)
+	numUsedResources := len(p.usedResources)
+	numAssignedResources := numUsedResources + numWarmResources + p.pendingCreate + len(p.coolDownQueue)
+
+	// warm pool is in draining state, set targets to zero
+	if p.warmPoolConfig.DesiredSize == 0 {
+		p.log.V(1).Info("DesiredSize is zero, warmPool is in draining state")
+		p.warmPoolConfig.WarmIPTarget = 0
+		p.warmPoolConfig.MinIPTarget = 0
+		p.warmPoolConfig.WarmPrefixTarget = 0
+	}
+
+	isMinIPTargetInvalid := p.warmPoolConfig.MinIPTarget < 0
+	isWarmIPTargetInvalid := p.warmPoolConfig.WarmIPTarget < 0
+	// Handle scenario where MinIPTarget is configured to negative integer which is invalid
+	if isMinIPTargetInvalid {
+		p.log.V(1).Info(
+			"MinIPTarget value is invalid negative integer, setting MinIPTarget to default",
+			"IPv4DefaultWinMinIPTarget", config.IPv4DefaultWinMinIPTarget,
+		)
+		p.warmPoolConfig.MinIPTarget = config.IPv4DefaultWinMinIPTarget
+	}
+	// Handle scenario where WarmIPTarget is configured to negative integer which is invalid
+	if isWarmIPTargetInvalid {
+		p.log.V(1).Info(
+			"WarmIPTarget value is invalid negative integer, setting warmIPTarget to default",
+			"IPv4DefaultWinWarmIPTarget", config.IPv4DefaultWinWarmIPTarget,
+		)
+		p.warmPoolConfig.WarmIPTarget = config.IPv4DefaultWinWarmIPTarget
+	}
+
+	availableResources := numWarmResources + p.pendingCreate - p.pendingDelete
+
+	// Calculate how many IPs we're short of the warm target
+	resourcesShort := max(p.warmPoolConfig.WarmIPTarget-availableResources, 0)
+
+	// Adjust short based on the minimum IP target
+	resourcesShort = max(resourcesShort, p.warmPoolConfig.MinIPTarget-numAssignedResources)
+
+	// Calculate how many IPs we're over the warm target
+	resourcesOver := max(availableResources-p.warmPoolConfig.WarmIPTarget, 0)
+
+	// Adjust over to not go below the minimum IP target
+	resourcesOver = max(min(resourcesOver, numAssignedResources-p.warmPoolConfig.MinIPTarget), 0)
+
+	// The final deviation is the difference between short and over
+	deviation := resourcesShort - resourcesOver
+
+	p.log.V(1).Info(
```

Comment:
I feel instead logging those important settings we can emit them as prometheus gauge metrics.

Reply:
I appreciate the suggestion, I actually think its a good suggestion as it would improve observability of the configuration value settings. However this would be additional work that I would need to thoroughly test and given my current time constraints I may not be able to get to this immediately. May I please suggest adding this suggestion as a technical debt item in the backlog to tackle at a later time?

Reply:
+1 for adding it as Prometheus metrics. If you want to create a backlog and make the changes in a follow-up PR, that would be appreciated, thanks!

Reply:
@sushrk Thanks for the feedback. I created this issue to track the work and am willing to take it up https://github.com/aws/amazon-vpc-resource-controller-k8s/issues/456